src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/ExpressionRewriterWithInitialContext.cs

```diff
@@ -0,0 +1,12 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Expressions
+{
+    public abstract class ExpressionRewriterWithInitialContext<TContext> : ExpressionRewriter<TContext>, IExpressionVisitorWithInitialContext<TContext, Expression>
+    {
+        public abstract TContext InitialContext { get; }
```

Comment:
Any reason why this couldn't just be `public virtual TContext InitialContext => default` and get rid of `ExpressionRewriterWrithDefaultInitialContext`?